packages/lib-react-components/src/Markdownz/Markdownz.js

```diff
@@ -66,6 +66,36 @@ class Markdownz extends React.Component {
     return null
   }
 
+  findResizedImages (children) {
+    const imageRegex = /(?:!\[(.*?)\]\((.*?)\))/g
+    const images = Array.from(children.matchAll(imageRegex))
```

Comment:
Not sure if you saw this or can see this, but github is warning:
> Polynomial regular expression used on uncontrolled data
This regular expression that depends on library input may run slow on strings starting with '!\[' and with many repetitions of '!\[a'.
This regular expression that depends on library input may run slow on strings starting with '!\[\](' and with many repetitions of '\](a'. 

https://github.com/zooniverse/front-end-monorepo/security/code-scanning/29?query=ref%3Arefs%2Fpull%2F2153%2Fhead

So, we need to rethink this particular regex.

Reply:
(This may still be the best way to do this, but I'm no regex expert, so will do some research into this)

Reply:
Maybe limit the number of captures this regex can make? I’m not sure of the best solution either but we do know that a markdown string won’t have an infinite number of images.

Reply:
Perhaps add the URL protocol (Question: should we restrict content to https://? I'd vote yes.) and a capture group that grabs the URL. Here's a very quick, rough example: https://regexr.com/5t02b

The URL is captured as group 1, so I think you should be able to get it using backreferences.
![Screenshot of the regexr.com inspector tool, showing a regext which captures only the URL.](https://user-images.githubusercontent.com/59547/118368813-75bf6280-b59a-11eb-9c7a-e1c368083edc.png)

Reply:
Here's an example which splits the capture into groups for the image URL and the size string, which might be more useful. https://regexr.com/5t05e

These examples might still trigger the GitHub warning, though. I'm not 100% sure how to fix that.

![Screenshot of a regex that combines the image and size regexes into a single expression. Examples of the capture group strings are also shown.](https://user-images.githubusercontent.com/59547/118372024-3f3c1480-b5a7-11eb-9d3a-168563eb28af.png)

Reply:
That regex doesn't match this image, so it might still need work.
```markdown
![This is a test image to try out regex matching. It has a couple of sentences in its description.](https://panoptes-uploads.zooniverse.org/production/subject_location/66094a64-8823-4314-8ef4-1ee228e49470.jpeg =100x)
```

Reply:
Changing the second number match to `\d*?` would fix that, I think.

Reply:
That expensive regex is gone from the code now.